### PR TITLE
[EMCAL-565]: Add deployment-ready bad channel map code.

### PR DESCRIPTION
**Pull request by hjbossi**

### CODE REVIEW

**Comment by hjbossi** [comment]
@mfasDa @jokonig

File: Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h [hunk 1]
```diff
@@ -74,32 +81,131 @@ class EMCALCalibExtractor
...
+        LOG(debug) << "Cell " << cellID << " is dead.";
         mOutputBCM.addBadChannel(cellID, o2::emcal::BadChannelMap::MaskType_t::DEAD_CELL);
-      } else if (E < maxVal || E > minVal) {
-        mOutputBCM.addBadChannel(cellID, o2::emcal::BadChannelMap::MaskType_t::GOOD_CELL);
       } else {
-        mOutputBCM.addBadChannel(cellID, o2::emcal::BadChannelMap::MaskType_t::BAD_CELL);
+        bool failed = false;
+        for (auto& [index, slice] : slices) {
+          auto ranges = calibrationInformation.goodCellWindowMap[index];
+          auto meanPerCell = calibrationInformation.energyPerHitMap[index][cellID];
```
Comment (mfasDa) [inline Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h]: Be careful with variable names (`index` in this case) - not that you redefine existing names

File: Detectors/EMCAL/calibration/run/runCalibOffline.cxx [hunk 2]
```diff
@@ -147,6 +153,12 @@ int main(int argc, char** argv)
     exit(2);
   }
 
+  if (debugMode) {
+    FairLogger::GetLogger()->SetLogScreenLevel("DEBUG");
+  } else {
+    FairLogger::GetLogger()->SetLogScreenLevel("INFO");
+  }
+
```
Comment (mfasDa) [inline Detectors/EMCAL/calibration/run/runCalibOffline.cxx]: Needs new fair logger:

``` c++
 if (debugMode) {
      fair::Logger::SetConsoleSeverity("debug");
} else {
      fair::Logger::SetConsoleSeverity("info");
}
```

File: Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALChannelCalibrator.h [hunk 3]
```diff
@@ -124,7 +124,9 @@ void EMCALChannelCalibrator<DataInput, DataOutput, HistContainer>::finalizeSlot(
 
   std::map<std::string, std::string> md;
   if constexpr (std::is_same<DataInput, o2::emcal::EMCALChannelData>::value) {
+    LOG(info) << "Launching the calibration.";
     auto bcm = mCalibrator->calibrateBadChannels(c->getHisto());
+    LOG(info) << "Done with the calibraiton";
```
Comment (mfasDa) [inline Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALChannelCalibrator.h]: Please set the debug statements to `LOG(debug)`

**Review by mfasDa — CHANGES_REQUESTED** [review]
Few small changes needed.

Before merging:
1) rebase to upstream/dev
2) squash all new commits of the branch into one
3) Change the commit message, be more detailed about what you changed, use commit header as short description and the message body for a more detailed explanation.

**Comment by hjbossi** [comment]
@mfasDa - the requested changes have been made. Should be good to merge.

**Review by mfasDa — APPROVED** [review]
Looks good

**Comment by mfasDa** [comment]
Finally the CI finished - forwarding to @shahor02, @davidrohr or @chiarazampolli for final merge.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**